### PR TITLE
code optimization

### PR DESCRIPTION
**Pull request by testwill**
io/ioutil deprecated etc.

### CODE REVIEW

**Review by ozkansen — APPROVED** [review]

File: mock/mock.go [hunk 1]
```diff
@@ -455,7 +455,7 @@ func (m *Mock) Called(arguments ...interface{}) Arguments {
 	// For Ex:  github_com_docker_libkv_store_mock.WatchTree.pN39_github_com_docker_libkv_store_mock.Mock
 	// uses interface information unlike golang github.com/docker/libkv/store/mock.(*Mock).WatchTree
 	// With GCCGO we need to remove interface information starting from pN<dd>.
-	re := regexp.MustCompile("\\.pN\\d+_")
+	re := regexp.MustCompile(`\.pN\d+_`)
```
Comment (dolmen) [inline mock/mock.go:458]: This change is not needed. This is just a quotes changes. While I agree that it improves redability, it has nothing to do in a MR that claims "code optimization". Please make a separate MR.

File: mock/mock_test.go [hunk 2]
```diff
@@ -1939,20 +1937,20 @@ func TestLoggingAssertExpectations(t *testing.T) {
 	AssertExpectationsForObjects(tcl, m, new(TestExampleImplementation))
 
 	require.Equal(t, 1, len(tcl.errs))
-	assert.Regexp(t, regexp.MustCompile("(?s)FAIL: 0 out of 1 expectation\\(s\\) were met.*The code you are testing needs to make 1 more call\\(s\\).*"), tcl.errs[0])
+	assert.Regexp(t, regexp.MustCompile(`(?s)FAIL: 0 out of 1 expectation\(s\) were met.*The code you are testing needs to make 1 more call\(s\).*`), tcl.errs[0])
```
Comment (dolmen) [inline mock/mock_test.go:1940]: This changes quoting. This is an unrelated change.

File: mock/mock_test.go [hunk 3]
```diff
@@ -1939,20 +1937,20 @@ func TestLoggingAssertExpectations(t *testing.T) {
...
 	require.Equal(t, 2, len(tcl.logs))
-	assert.Regexp(t, regexp.MustCompile("(?s)FAIL:\tGetTime\\(int\\).*"), tcl.logs[0])
+	assert.Regexp(t, regexp.MustCompile(`(?s)FAIL:\tGetTime\(int\).*`), tcl.logs[0])
```
Comment (dolmen) [inline mock/mock_test.go:1942]: This changes quoting. This is an unrelated change.

File: suite/suite_test.go [hunk 4]
```diff
@@ -429,7 +429,7 @@ func (sc *StdoutCapture) StopCapture() (string, error) {
 	}
 	os.Stdout.Close()
 	os.Stdout = sc.oldStdout
-	bytes, err := ioutil.ReadAll(sc.readPipe)
+	bytes, err := io.ReadAll(sc.readPipe)
```
Comment (dolmen) [inline suite/suite_test.go:432]: This change requires Go 1.16. `testify` is used in a wide range of project. This is not a good thing to force users to upgrade Go while it is not necessary (`io/ioutil` will stay forever).

**Review by dolmen — CHANGES_REQUESTED** [review]

**Comment by dolmen** [comment]
This PR bundles too many unrelated changes with a misleading title. I recommend @testwill to close it and submit more targetted changes and take more care in the description of the PR.

**Pull request closed without merging**